client/common/client.c

```diff
@@ -934,6 +937,35 @@ BOOL client_cli_present_gateway_message(freerdp* instance, UINT32 type, BOOL isD
 	return TRUE;
 }
 
+BOOL client_cli_get_aad_auth_code(const char* hostname, char** code)
```

Comment:
Might be a good idea to add `rdpContext` as argument so you can get/set data for the session in the callback